core/src/main/java/com/alibaba/nacos/core/listener/StartingApplicationListener.java

```
@@ -128,13 +134,38 @@ private void injectEnvironment(ConfigurableEnvironment environment) {
     
     private void loadPreProperties(ConfigurableEnvironment environment) {
         try {
-            environment.getPropertySources().addLast(new OriginTrackedMapPropertySource(FIRST_PRE_PROPERTIES,
-                    EnvUtil.loadProperties(EnvUtil.getApplicationConfFileResource())));
+            SOURCES.putAll(EnvUtil.loadProperties(EnvUtil.getApplicationConfFileResource()));
+            environment.getPropertySources()
+                    .addLast(new OriginTrackedMapPropertySource(NACOS_APPLICATION_CONF, SOURCES));
+            registerWatcher();
         } catch (IOException e) {
             throw new NacosRuntimeException(NacosException.SERVER_ERROR, e);
         }
     }
     
+    private void registerWatcher() {
+        try {
+            WatchFileCenter.registerWatcher(EnvUtil.getConfPath(), new FileWatcher() {
+                @Override
+                public void onChange(FileChangeEvent event) {
+                    try {
+                        Map<String, ?> tmp = EnvUtil.loadProperties(EnvUtil.getApplicationConfFileResource());
+                        SOURCES.putAll(tmp);
+                    } catch (IOException ignore) {
+                    
```

Why just ignore the Exception ?